### ISSUE TITLE
ClientConfig::create does not throw error if kafka topic is not reachable

### ISSUE THREAD

**Issue by surajpayu** [fede1024/rust-rdkafka#112]
I mistakenly gave wrong ip address of kafka. But below line of code got executed successfully without giving any error
```
 let producer = ClientConfig::new()
        .set("bootstrap.servers", brokers)
            .set("produce.offset.report", "true")
        .create::<FutureProducer<_>>()
.expect("Producer creation error");
```

It got detected only once i did producer.send(...). I am trying to do send inside a mpsc channel and thus want to detect whether producer is connected or not at the time of creating it. How should i go about it?

**Comment by benesch** [comment]
The underlying librdkafka library sends error messages about failing to connect to the specified broker(s) to a global error handler, [`ClientContext::error`](https://docs.rs/rdkafka/0.21.0/rdkafka/client/trait.ClientContext.html#method.error). You can provide a custom context implementation that overrides this method and handles the errors appropriately.

The default error handler just logs them via `log::error!`, so if you have logging enabled you'll see something like

```
12:07:36.334 (t: producer polling thread) ERROR - rdkafka::client - librdkafka: Global error: Resolve (Local: Host resolution failure): foo:9092/bootstrap: Failed to resolve 'foo:9092': nodename nor servname provided, or not known (after 1008ms in state CONNECT)
12:07:36.335 (t: producer polling thread) ERROR - rdkafka::client - librdkafka: Global error: AllBrokersDown (Local: All broker connections are down): 1/1 brokers are down
```

Either of those errors would be reasonable to look for, though the first is most likely to be the result of a mistyped IP. The error types are `KafkaError::Global(RDKafkaError::Resolve)` and `KafkaError::Global(RDKafkaError::AllBrokersDown)`.

Feel free to open a new issue if you have trouble with this approach.
